nat/traversal/pinger.go

```diff
@@ -95,9 +97,15 @@ type Params struct {
 func (p *Pinger) Start() {
 	log.Info(prefix, "Starting a NAT pinger")
 
-	// We dont need to run pinger if NAT port auto-configuration is successful
-	if p.natEventWaiter.WaitForEvent().Type == SuccessEventType {
+	resultChannel := make(chan bool)
```

Comment:
We need to have it buffered to finish goroutine even if the stop was invoked early. 
```suggestion
	resultChannel := make(chan bool, 1)
```